keep-ui/app/settings/add-user-modal.tsx

```diff
@@ -0,0 +1,237 @@
+import { Dialog } from "@headlessui/react";
+import { useState } from "react";
+import {
+  TextInput,
+  Button,
+  Subtitle,
+  Icon,
+  SearchSelect,
+  SearchSelectItem,
+} from "@tremor/react";
+import { AuthenticationType } from "utils/authenticationType";
+import { User } from "./models";
+import { getApiURL } from "utils/apiUrl";
+import { InfoCircledIcon } from "@radix-ui/react-icons";
+import "./add-user-modal.css";
+
+interface RoleOption {
+  value: string;
+  label: string | JSX.Element;
+  tooltip?: string;
+  isDisabled?: boolean;
+}
+
+interface AddUserModalProps {
+  isOpen: boolean;
+  onClose: () => void;
+  authType: string;
+  setUsers: React.Dispatch<React.SetStateAction<User[]>>;
+  accessToken: string;
+}
+
+const roleOptions: RoleOption[] = [
+  {
+    value: "admin",
+    label: "Admin",
+    tooltip: "Admin has read/write/update/delete for every resource",
+  },
+  {
+    value: "noc",
+    label: "NOC",
+    tooltip: "NOC has the ability to view and assign alerts",
+  },
+  {
+    value: "create_new",
+    label: "Create custom role",
+    isDisabled: true,
+    tooltip: "For custom roles, contact Keep team",
+  },
+];
+
+export default function AddUserModal({
+  isOpen,
+  onClose,
+  authType,
+  setUsers,
+  accessToken,
+}: AddUserModalProps) {
+  const [email, setEmail] = useState("");
+  const [emailError, setEmailError] = useState("");
+  const [selectedRole, setSelectedRole] = useState<string>();
+  const [password, setPassword] = useState("");
+  const [addUserError, setAddUserError] = useState("");
```

Comment:
Using an [uncontrolled form](https://twitter.com/JoshWComeau/status/1330302183431147526) would shorten this code

Reply:
Or you plan on adding more forms with validation, I can recommend a good form library: [react-hook-form](https://react-hook-form.com/)

Reply:
Done